File: app/src/main/java/com/sorbonne/atom_d/tools/CustomRecyclerView.java
package com.sorbonne.atom_d.tools;

import android.annotation.SuppressLint;
import android.content.Context;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import org.jetbrains.annotations.NotNull;

@SuppressLint("ViewConstructor")
public class CustomRecyclerView extends RecyclerView{


    public enum CustomLayoutManager{
        LINEAR_LAYOUT
    }

    RecyclerView mRecyclerView;
    Adapter<?> mAdapter;
    LayoutManager mLayout;


    public CustomRecyclerView(@NonNull @NotNull Context context, RecyclerView recyclerView, Adapter<?> adapter, CustomLayoutManager layout) {
        super(context);
        mRecyclerView = recyclerView;
        mAdapter = adapter;
        switch (layout){
            case LINEAR_LAYOUT:
                mLayout = new LinearLayoutManager(context);
                break;
        }
    }

    public RecyclerView getRecyclerView()
    {
        mRecyclerView.setAdapter(mAdapter);
        mRecyclerView.setLayoutManager(mLayout);

        return mRecyclerView;
    }
}

File: app/src/main/java/com/sorbonne/atom_d/tools/CustomRecyclerView.kt
package com.sorbonne.atom_d.tools

import android.content.Context
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.Adapter
import androidx.recyclerview.widget.RecyclerView.LayoutManager

class CustomRecyclerView(context: Context, private val recyclerView: RecyclerView, private val adapter: Adapter<*>, layout: CustomLayoutManager) {
    enum class CustomLayoutManager{
        LINEAR_LAYOUT
    }

    private lateinit var layoutManager: LayoutManager

    init {
        if(layout == CustomLayoutManager.LINEAR_LAYOUT){
            layoutManager = LinearLayoutManager(context)
        }
    }

    fun getRecyclerView(): RecyclerView {
        recyclerView.adapter = adapter
        recyclerView.layoutManager = layoutManager
        return recyclerView
    }

}